codegen/csg.adoc

```diff
@@ -8,4 +8,193 @@ permalink: /codegen_custom_swagger_gentemplate/
+will use are:
+
+`-o dir`:: Specifies the directory where the skeleton project will be created; default is current directory.
+
+`-n name`:: Specifies the name for the new SCG module, which is put to many uses in the skeleton,
+some of which imply restrictions on the name that are not enforced by SCG. These
+include:
+
+** It is the advertised name of the module (i.e. the value its `CodegenConfig#getName()` method
+   returns). The CLI `generate` command's `-l` (`--lang`) option maps to these names.
+** It is incorporated into the name of the generator class, which is the name value with its first
+   character upper-cased, concatenated with `Generator`. E.g. `-n myModule` would generate a class
+   named `MyModuleGenerator`.
+** It is incorporated into the advertised description of the model (the value returned by
+   `CodegenConfig#getHelp()).
+** It is incorporated into the name of the default output folder used by the module. (This is
+   unused when the module is executed in {RAS} as a GenTemplate, since the output directory is in
+   that case specified in the GenTarget.)
+** It is used as the template directory name. This directory will appear as a top-level Java
+   resource directory in the generated project.
+
+`-p package`:: Specifies the package in which the generator class will reside. The default is
+`io.swagger.codegen`.
+
+In preparation for this step we need to install the Swagger Codegen Client jar file. One way to do
+that is to clone the link:https://github.com/swagger-api/swagger-codegen[swagger-api/swagger-codegen
+git repo], checking out version 2.1.4 (using the git tag `v2.1.4`) and then running `mvn package`
+from the working tree root. This will build the needed jar file in
+[.nowrap]`modules/swagger-codegen-cli/target/swagger-codegen-cli.jar`.
+
+We'll execute this step by starting a command shell for our system and navigating to a directory
+that will hold the skeleton as a subdirectory. For simplicity we'll assume that parent directory is
+the user's home directory, designated in the linux style as `~`, and that the swagger-codegen git
+clone is located in `~/git/swagger-codegen`.
+
+After all the foregoing, this step finally comes down to a single command, which will create our
+skeleton:
+
+[source%nowrap]
+--
+java -jar ~/git/swagger-codegen/modules/swagger-codegen-cli/target/swagger-codegen-cli.jar meta -o ~/LexiconSCGentemplates -n lexiconSCGenTemplate -p my.org
+--
+
+=== Importing the Skeleton into {RAS}
+
+The skeleton is configured as a Maven project, and {RAS} includes the capability of importing Maven
+projects into the workspace. From the menu bar, select _File -> Import..._, then select _Maven ->
+Existing Maven Projects_ in the dialog that appears. Click _Next_.
+
+image::import-maven-step-1.png[Import Skeleton Project Step 1,role=text-center]
+
+Browse to the output folder you specified for the SCG `meta` command, and you should see the Maven
```

Comment:
I think this step assumes you've placed the output somewhere within the workspace.  It wouldn't hurt to reiterate that point here.  What happens if you try to import something that's outside the workspace?  Is there an additional step to link it in...?